[Instrumentation.Process] Update .NET API for virtual memory usage (#762)
open-telemetry/opentelemetry-dotnet-contrib: src/OpenTelemetry.Instrumentation.Process/README.md

### process.memory.virtual

> All the address space this process can read from without trigerring an access

Reviewer: ```suggestion
The amount of committed virtual memory for this process.
One way to think of this is all the address space this process can read from without trigerring an access
```

that cannot be shared with other processes.
All the address space this process can read from without trigerring an access
violation; including memory backed solely by RAM, pagefile-backed memory and
file-backed memory.

| Units | Instrument Type         | Value Type |
|-------|-------------------------|------------|
|  `By` | ObservableUpDownCounter | `Double`   |

The API used to retrieve the value is:

> * [Process.WorkingSet64](https://learn.microsoft.com/dotnet/api/system.diagnostics.process.privatememorysize64):

Author: Note: this was an copy-paste error in the original document.

### process.memory.virtual

The amount of committed virtual memory for this process.
One way to think of this is all the address space this process can read from

> without trigerring an access violation; including memory backed solely by RAM,

Reviewer 2: ```suggestion
without trigerring an access violation; this includes memory backed solely by RAM,
```

> pagefile-backed memory and file-backed memory.

Reviewer 2: by a swapfile/pagefile and by other mapped files on disk.

> pagefile-backed memory and file-backed memory.

Reviewer 2: ```suggestion
by a swapfile/pagefile and by other mapped files on disk.
```
